pkg/interface/src/views/apps/links/components/lib/link-submit.tsx

```diff
@@ -0,0 +1,253 @@
+
+    if (linkValid) {
+      if (hasProvider(linkValue)) {
+        fetch(`https://noembed.com/embed?url=${linkValue}`)
+        .then(response => response.json())
+        .then((result) => {
+          if (result.title) {
+            this.setState({ linkTitle: result.title });
+          }
+        }).catch((error) => {/*noop*/});
+      } else {
+        this.setState({
+          linkTitle: decodeURIComponent(linkValue
+            .split('/')
+            .pop()
+            .split('.')
+            .slice(0, -1)
+            .join('.')
+            .replace('_', ' ')
+            .replace(/\d{4}\.\d{1,2}\.\d{2}\.\.\d{2}\.\d{2}\.\d{2}-/, '')
+            )
+        })
+      }
+    }
+  }
+
+  setLinkValue(event) {
+    this.setState({ linkValue: event.target.value });
+    this.setLinkValid(event.target.value);
+  }
+
+  setLinkTitle(event) {
+    this.setState({ linkTitle: event.target.value });
+  }
+
+  uploadSuccess(url) {
+    this.setState({ linkValue: url });
+    this.setLinkValid(url);
+  }
+
+  uploadError(error) {
+    //  no-op for now
+  }
+
+  readyToUpload(): boolean {
+    return Boolean(this.s3Uploader.current && this.s3Uploader.current.inputRef.current);
+  }
+
+  onDragEnter() {
+    if (!this.readyToUpload()) {
+      return;
+    }
+    this.setState({ dragover: true });
+  }
+
+  onDrop(event: DragEvent) {
+    this.setState({ dragover: false });
+    if (!event.dataTransfer || !event.dataTransfer.files.length) {
+      return;
+    }
+    event.preventDefault();
+    this.uploadFiles(event.dataTransfer.files);
+  }
+
+  onPaste(event: ClipboardEvent) {
+    if (!event.clipboardData || !event.clipboardData.files.length) {
+      return;
+    }
+    event.preventDefault();
+    event.stopPropagation();
+    this.uploadFiles(event.clipboardData.files);
+  }
+
+  uploadFiles(files: FileList) {
+    if (!this.readyToUpload()) {
+      return;
+    }
+    this.s3Uploader.current.inputRef.current.files = files;
+    const fire = document.createEvent("HTMLEvents");
+    fire.initEvent("change", true, true);
+    this.s3Uploader.current?.inputRef.current?.dispatchEvent(fire);
+  }
+
+  render() {
+    const activeClasses = (this.state.linkValid && !this.state.disabled)
+      ? 'green2 pointer' : 'gray2';
+
+    const focus = (this.state.submitFocus)
+      ? 'b--black b--white-d'
+      : 'b--gray4 b--gray2-d';
+
+    return (
+      <div
+        className={`relative ba br1 w-100 mb6 ${focus}`}
+        onDragEnter={this.onDragEnter.bind(this)}
+        onDragOver={e => {e.preventDefault();this.setState({ dragover: true})}}
+        onDragLeave={() => this.setState({ dragover: false })}
+        onDrop={this.onDrop}
+      >
+        {this.state.dragover ? <SubmitDragger /> : null}
+        <div className="relative">
+          {(this.state.linkValue || this.state.urlFocus || this.state.disabled) ? null : <span className="gray2 absolute pl2 pt4 pb2 f8" style={{pointerEvents: 'none'}}>
```

Comment:
```suggestion
          {(this.state.linkValue || this.state.urlFocus || this.state.disabled) ? null : <span className="gray2 absolute pl2 pt3 pb2 f8" style={{pointerEvents: 'none'}}>
```

Dropped this one sorry